core/src/main/java/org/springframework/security/authorization/method/DefaultPostInvocationAuthorizationDeniedPostProcessor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method;
+
+import org.springframework.security.authorization.AuthorizationException;
+import org.springframework.security.authorization.AuthorizationResult;
+
+public class DefaultPostInvocationAuthorizationDeniedPostProcessor
+		implements AuthorizationDeniedPostProcessor<MethodInvocationResult> {
```

Comment:
I don’t think the class can be generic because it is used inside the PreAuthorize and PostAuthorize classes and annotation’s attributes must be constants. We need an actual implementation of the concrete type to use